app/screens/earns-screen/earns-quiz.tsx

```diff
@@ -178,15 +178,48 @@ gql`
   }
 `
 
-export const EarnQuiz = ({ route, navigation }: Props) => {
-  const { title, text, amount, answers, feedback, question, id, completed } = route.params
+export const EarnQuiz = ({ route }: Props) => {
+  const { LL } = useI18nContext()
+  const quizQuestionsContent = getQuizQuestionsContent({ LL })
+  const navigation = useNavigation<StackNavigationProp<RootStackParamList, "earnsQuiz">>()
+
+  const [permutation] = useState<ZeroTo2[]>(shuffle([0, 1, 2]))
+
+  const { data } = useMyQuizQuestionsQuery()
+  const myQuizQuestions = data?.me?.defaultAccount?.quiz?.slice() ?? []
+
+  const { id } = route.params
+
+  const allCards = React.useMemo(
+    () => quizQuestionsContent.map((item) => item.content).flatMap((item) => item),
+    [quizQuestionsContent],
+  )
+
+  const cardNoMetadata = React.useMemo(
+    () => allCards.find((item) => item.id === id),
+    [allCards, id],
+  )
+
+  if (!cardNoMetadata) {
+    // should never happen
+    throw new Error("card not found")
+  }
+
+  const card = augmentCardWithGqlData({ card: cardNoMetadata, myQuizQuestions })
+  const { title, text, amount, answers, feedback, question, completed } = card
 
   const [quizCompleted] = useQuizCompletedMutation()
-  const [isCompleted, setIsCompleted] = useState(completed)
+  const [isCompleted, setIsCompleted] = useState(false)
   const [quizVisible, setQuizVisible] = useState(false)
   const [recordedAnswer, setRecordedAnswer] = useState<number[]>([])
-  const [permutation] = useState<ZeroTo2[]>(shuffle([0, 1, 2]))
-  const { LL } = useI18nContext()
+
+  // when apollo loading = true, completed value will be false
+  // so we have to wait for the loading to complete
+  React.useEffect(() => {
```

Comment:
Not sure I understand why this is necessary.

Reply:
because if we have that:
`const [isCompleted, setIsCompleted] = useState(completed)`

and completed is derived from:
`data?.me?.defaultAccount?.quiz?.slice() ?? []`

and is `false` if data is null, then `completed` will always be initialized to false. 

because on the first run, apollo client will return `{loading: true, data: undefined}` (even if the data is cached)

Reply:
That makes sense for that change, but is `isCompleted` necessary at all or could we just use `completed`

Reply:
I think you are right actually. I just re-use `  const [isCompleted, setIsCompleted] = useState(completed)` because it was here already but didn't think about it. 

maybe let's merge this and I'll do that in a follow up PR?

Reply:
Sounds good